File: java/src/main/java/com/alxarafe/app/application/catalogue/ListUoms.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.valueobject.Uom;

import java.util.List;

public final class ListUoms {
    private final UomRepository repository;

    public ListUoms(UomRepository repository) {
        this.repository = repository;
    }

    public List<Uom> execute() {
        return repository.findAll();
    }
}

File: php/src/Application/Catalogue/ListUoms.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\ValueObject\Uom;

final readonly class ListUoms
{
    public function __construct(private UomRepository $repository)
    {
    }

    /** @return list<Uom> */
    public function execute(): array
    {
        return $this->repository->findAll();
    }
}
